packages/medusa/src/repositories/product-category.ts

```diff
@@ -1,5 +1,47 @@
-import { EntityRepository, TreeRepository } from "typeorm"
+import { EntityRepository, TreeRepository, Brackets, ILike } from "typeorm"
 import { ProductCategory } from "../models/product-category"
+import { ExtendedFindConfig, Selector } from "../types/common"
 
 @EntityRepository(ProductCategory)
-export class ProductCategoryRepository extends TreeRepository<ProductCategory> {}
+export class ProductCategoryRepository extends TreeRepository<ProductCategory> {
+  public async getFreeTextSearchResultsAndCount(
+    options: ExtendedFindConfig<ProductCategory, Selector<ProductCategory>> = {
+      where: {},
+    },
+    q: string | undefined
+  ): Promise<[ProductCategory[], number]> {
+    const options_ = { ...options }
+
+    const queryBuilder = this.createQueryBuilder("product_category")
+      .select()
+      .skip(options.skip)
+      .take(options.take)
+
+    if (q) {
+      delete options_?.where?.name
+      delete options_?.where?.handle
+
+      queryBuilder.where(
+        new Brackets((bracket) => {
+          bracket
+            .where({ name: ILike(`%${q}%`) })
+            .orWhere({ handle: ILike(`%${q}%`) })
+        })
+      )
+    }
+
+    queryBuilder.andWhere(options.where)
+
+    if (options.withDeleted) {
```

Comment:
todo: options_ for consistency, eveything should be based on the copy since this is the one that can be altered. otherwise we might break thinks easily if we change something and the wrong options object is used :)